### PR TITLE
Offset vector to replace normal in plane sampler

### PR DESCRIPTION
**Pull request by gyalla**
The latest version of AMR-Wind replaces "normal" with "offset_vector" in the plane sampler descriptions. @lawrenceccheung, I'm starting this PR to update the frontend to reflect this change. Do we want to preserve any legacy functionality for "normal" or just move everything to offset_vector as I've tried to do here?

### CODE REVIEW

File: plotfunctions.py [hunk 1]
```diff
@@ -288,7 +288,12 @@ def plotDomain(self, ax=None, verbose=False, plotskip=1):
                     offsets =[float(x) for x in allpdict[p+'.offsets'].split()]
                 else:
                     offsets = [0.0]
-                offsetnormal = np.array(allpdict[p+'.normal'])
+                # GRY: Using try - except here to start. Is the default value None if the field does not exists though?
+                try:
+                    offsetnormal = np.array(allpdict[p+'.offset_vector'])
+                except:
+                    print("offset_vector not specified. Using legacy interface: .normal")
+                    offsetnormal = np.array(allpdict[p+'.normal'])
```
Comment (gyalla) [inline plotfunctions.py]: @lawrenceccheung , is there a default value if keys do not exist in allpdict? If so, I can revert this to an if else statement

Comment (lawrenceccheung) [reply plotfunctions.py]: I believe the default value for items which were not specified by the user is what's given the `config.yaml` `default` section, and if there's nothing there, then it's just `None`.

**Comment by lawrenceccheung** [comment]
Linking this to the PR https://github.com/Exawind/amr-wind/pull/1137.

**Review by lawrenceccheung — APPROVED** [review]

File: plotfunctions.py (hunk 1 above)
Comment (gyalla) [reply plotfunctions.py]: Got it, OK, the default for normal and offset_vector is [0,0,0]. In that case, this "try, except" block should be correct. For new setups, offset_vector will be read in for plotting/validating inputs. However, if it doesn't exist, normal will be read instead. For the actual case setups, both offset_vector and normal will be written to maintain compatibility with either amr-wind builds.

**Pull request merged**